sycl/include/sycl/ext/intel/esimd/memory.hpp

```diff
@@ -4561,17 +4562,17 @@ atomic_update(T *p, simd_view<Toffset, OffsetRegionTy> offsets,
 /// @return A vector of the old values at the memory locations before the
 ///   update.
 ///
-template <atomic_op Op, typename T, int N, typename Toffset,
+template <atomic_op Op, typename T, int N, typename Toffset, typename ViewedTy,
           typename OffsetRegionTy = region1d_t<Toffset, N, 1>,
-          typename RegionTy = region1d_t<T, N, 1>,
+          typename RegionTy = region1d_t<ViewedTy, N, 1>,
```

Comment:
All such places do not require the default value for template parameter. That default value is never used and is always set by the type of `byte_offset` and `src` operands.